backend/coins/eth/blockbook/blockbook.go

```diff
@@ -0,0 +1,163 @@
+// Copyright 2025 Shift Crypto AG
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package blockbook
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"io"
+	"math/big"
+	"net/http"
+	"net/url"
+	"path"
+
+	"github.com/BitBoxSwiss/bitbox-wallet-app/backend/accounts"
+	"github.com/BitBoxSwiss/bitbox-wallet-app/backend/coins/eth/erc20"
+	"github.com/BitBoxSwiss/bitbox-wallet-app/backend/coins/eth/rpcclient"
+	ethtypes "github.com/BitBoxSwiss/bitbox-wallet-app/backend/coins/eth/types"
+	"github.com/BitBoxSwiss/bitbox-wallet-app/util/errp"
+	"github.com/ethereum/go-ethereum"
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/ethereum/go-ethereum/core/types"
+	"golang.org/x/time/rate"
+)
+
+// callsPerSec is the number of blockbook requests allowed
+// per second.
+// TODO bznein determine a better value for this.
+const callsPerSec = 3.8
+
+// Blockbook is a rate-limited blockbook api client.
+type Blockbook struct {
+	url        string
+	httpClient *http.Client
+	limiter    *rate.Limiter
+}
+
+// NewBlockbook creates a new instance of EtherScan.
+func NewBlockbook(chainId string, httpClient *http.Client) *Blockbook {
+	// TODO chainID is not used here, do we have blockbook running for SEPETH as well?
```

Comment:
Best to panic if chainID is not `1` until you find out, so it can never be used on the wrong chain.

Reply:
Good idea, done